lib/executor/constant_arrival_rate.go

```diff
@@ -141,12 +147,12 @@ func (carc ConstantArrivalRateConfig) Validate() []error {
 // maximum waiting time for any iterations to gracefully stop. This is used by
 // the execution scheduler in its VU reservation calculations, so it knows how
 // many VUs to pre-initialize.
-func (carc ConstantArrivalRateConfig) GetExecutionRequirements(es *lib.ExecutionSegment) []lib.ExecutionStep {
+func (carc ConstantArrivalRateConfig) GetExecutionRequirements(et *lib.ExecutionTuple) []lib.ExecutionStep {
 	return []lib.ExecutionStep{
 		{
 			TimeOffset:      0,
-			PlannedVUs:      uint64(es.Scale(carc.PreAllocatedVUs.Int64)),
-			MaxUnplannedVUs: uint64(es.Scale(carc.MaxVUs.Int64 - carc.PreAllocatedVUs.Int64)),
+			PlannedVUs:      uint64(et.ScaleInt64(carc.PreAllocatedVUs.Int64)),
+			MaxUnplannedVUs: uint64(et.ScaleInt64(carc.MaxVUs.Int64 - carc.PreAllocatedVUs.Int64)),
```

Comment:
I'm not sure this is correct. `et.ScaleInt64(carc.MaxVUs.Int64 - carc.PreAllocatedVUs.Int64)` is not necessarily the same as `et.ScaleInt64(carc.MaxVUs.Int64) - et.ScaleInt64(carc.PreAllocatedVUs.Int64)`, right? And don't we need the latter here?

Reply:
Yeah I also wondered ... apparently didn't leave a TODO though. We definitely want the latter, but I couldn't come up with an example where it breaks ... will try again as I fix the other things. 

This will need fixing in the variable arrival rate as well

Reply:
resolved in https://github.com/loadimpact/k6/pull/1323/commits/6a6cb4170079e2a80e1dd5be1e571fc03e708be2